frontend/src/components/RestaurantCard/RestaurantCard.tsx

```diff
@@ -1,22 +1,35 @@
 import { styled } from 'styled-components';
 import { BORDER_RADIUS, FONT_SIZE, truncateText } from '~/styles/common';
 import ProfileImage from '../@common/ProfileImage';
-import { Restaurant } from '~/@types/restaurant.types';
-import { Celeb } from '~/@types/celeb.types';
+import { BASE_URL } from '~/App';
+
+import type { Celeb } from '~/@types/celeb.types';
+import type { Restaurant } from '~/@types/restaurant.types';
 
 interface RestaurantCardProps {
   restaurant: Restaurant;
-  celebs: Celeb[];
-  size: number;
-  onClick: React.MouseEventHandler;
+  celebs?: Celeb[];
+  size?: number;
+  type?: 'list' | 'map';
+  onClick?: React.MouseEventHandler;
+  onMouseEnter?: (targetId: number) => void;
+  onMouseLeave?: () => void;
 }
 
-function RestaurantCard({ restaurant, celebs, size, onClick }: RestaurantCardProps) {
+function RestaurantCard({
+  restaurant,
+  celebs,
+  size,
+  type = 'list',
+  onClick = () => {},
+  onMouseEnter,
+  onMouseLeave,
+}: RestaurantCardProps) {
   const { images, name, roadAddress, category } = restaurant;
 
   return (
-    <StyledContainer onClick={onClick}>
-      <StyledImage alt={`${name} 대표 이미지`} src={`http://3.35.157.27:3000/images-data/${images[0].name}`} />
+    <StyledContainer onClick={onClick} onMouseEnter={() => onMouseEnter(restaurant.id)} onMouseLeave={onMouseLeave}>
```

Comment:
onMouseEnter 안의 함수도 밖으로 분리해 주실 수 있을까요?

```suggestion
    <StyledContainer onClick={onClick} onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave}>
```